gxutil/lckfile.go

```diff
@@ -0,0 +1,40 @@
+package gxutil
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+const LockVersion = 1
+
+type LockDep struct {
```

Comment:
Could we document what's the difference between a `LockDep` and a `Dependency`?

I mean, why are we using this new structure instead of extending `Dependency` or including it in this structure hierarchy? I'm not saying it should, but reading this automatically makes me thing of what we now call a dependency (inside the `package.json`), and now I'll be seeing a different file that also talks about dependencies but with a different structure which can be confusing.

I understand that conceptually both are referring to a package (that is a dep of another package) but with a different representation in a different context so it would be nice to make that link stronger.